test-suite/src/test/groovy/io/micronaut/EventListenerSpec.groovy

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut
+
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Requires
+import io.micronaut.context.event.ApplicationEvent
+import io.micronaut.context.event.ApplicationEventListener
+import io.micronaut.context.event.ShutdownEvent
+import io.micronaut.context.event.StartupEvent
+import io.micronaut.core.order.OrderUtil
+import io.micronaut.discovery.event.ServiceReadyEvent
+import io.micronaut.discovery.event.ServiceStoppedEvent
+import io.micronaut.inject.qualifiers.Qualifiers
+import io.micronaut.runtime.event.annotation.EventListener
+import io.micronaut.runtime.server.EmbeddedServer
+import io.micronaut.runtime.server.event.ServerShutdownEvent
+import io.micronaut.runtime.server.event.ServerStartupEvent
+import jakarta.inject.Singleton
+import spock.lang.RepeatUntilFailure
+import spock.lang.Specification
+import spock.util.concurrent.PollingConditions
+
+class EventListenerSpec extends Specification {
+
+    @RepeatUntilFailure(maxAttempts = 100) // Retry the test multiple times to test the listeners order
```

Comment:
https://spockframework.org/spock/docs/2.3/extensions.html
> @RepeatUntilFailure annotation can be used to repeat a single feature until it fails or until it reaches the maxAttempts value. This is intended to aid in manual debugging of flaky tests and should not be committed to source control.

Should not we remove `@RepeatUntilFailure` and don't commit it so that we don't run this test a hundred times? I assume you run it locally a 100 times and it passed always. Thus, it should not be flaky.

Reply:
We can, but then if you remove the order in the future it will become flaky instead of failing directly on some attempt

Reply:
I think it is better to become flaky if we break it and then we should fix it than run it 100 times.